wordpress/wp-content/themes/headless/setup/helpers/stellate.php

```diff
@@ -0,0 +1,145 @@
+<?php
+// Disable a plugin based on environment. We don't want to purge cache during local/staging development.
+// Also allows for listening to graphcdn_purge event to perform additional invalidation
+
+// add_action('admin_init', 'disable_stellate_plugin');
+
+// TODO: we want want our staging/dev environments hitting production, so we need to override the service name and token
+// $stellate_staging_service_name = "";
+// $stellate_staging_token = "";
+
+function disable_stellate_plugin()
+{
+  $plugin_slug = 'stellate/wp-stellate.php';
+  if (!defined('WP_STELLATE_ENABLED')) {
+    if (defined('WP_HOST') && WP_HOST == "production") {
+      global $stellate_production_enabled;
+      if ($stellate_production_enabled) {
+        define('WP_STELLATE_ENABLED', true);
+      }
+    }
+
+    if (defined('WP_HOST') && WP_HOST == "staging") {
+      global $stellate_staging_enabled;
+      if ($stellate_staging_enabled) {
+        define('WP_STELLATE_ENABLED', true);
+      }
+    }
+
+    if (defined('WP_HOST') && WP_HOST == "development") {
+      global $stellate_development_enabled;
+      if ($stellate_development_enabled) {
+        define('WP_STELLATE_ENABLED', true);
+      }
+    }
+  }
+
+
+  if (defined('WP_STELLATE_ENABLED') && WP_STELLATE_ENABLED) {
+    // leave enabled
+  } else {
+    // disable
+    deactivate_plugins($plugin_slug);
+  }
+}
+
+function purge_redirection()
+{
+  stellate_add_purge_entity('purged_types', 'RedirectionRedirects');
+}
+
+function purge_acf_options()
+{
+  stellate_add_purge_entity('purged_types', 'AcfOptionsThemeSettings');
```

Comment:
Should this be a variable that comes from functions.php? In the older version we had variables that enabled redirection (maybe someone is using yoast seo instead), and in this case maybe in a different version of ACF the variable changes. Ideally our per site config vars in functions.php mean this file can always be updated from main without breaking anything across sites.

Reply:
updated